src/bin/sccache-dist/cmdline/parse.rs

```diff
@@ -0,0 +1,369 @@
+
+        for invalid_val in invalid_vals {
+            let args = auth_generate_shared_tokens_bits(invalid_val);
+            assert!(try_parse_from(args).is_err());
+        }
+    }
+
+    #[test]
+    fn auth_generate_server_token_needs_key_source() {
+        let server_args = &["--server", "127.0.0.1:4321"];
+
+        let no_key = auth_generate_jwt_hs256_server_token(server_args);
+        assert!(try_parse_from(no_key).is_err());
+
+        let mut too_many_keys = auth_generate_jwt_hs256_server_token(server_args);
+        too_many_keys.extend_from_slice(&["--secret-key", "secret", "--config", "some/path.toml"]);
+        assert!(try_parse_from(too_many_keys).is_err());
+    }
+
+    // This is all just to work around `PartialEq` not being on some of the values used in variants
+    // for `Command` yet
+    fn assert_args_parse_to_auth(args: Vec<&'static str>, ideal_auth: AuthSubcommand) {
+        match try_parse_from(&args) {
+            Ok(Command::Auth(auth)) => assert_eq!(auth, ideal_auth),
+            _ => panic!("Bad parsing for: {:#?}", args),
+        }
+    }
+
+    #[test]
+    fn auth_generate_jwt_hs256_key_good() {
+        let args = vec![EXE, "auth", "generate-jwt-hs256-key"];
+
+        assert_args_parse_to_auth(args, AuthSubcommand::Base64 { num_bytes: 256 / 8 });
+    }
+
+    #[test]
+    fn auth_generate_jwt_hs256_server_token_good() {
+        let base = auth_generate_jwt_hs256_server_token(&["--server", "127.0.0.1:4321"]);
+        let server_socket: SocketAddr = "127.0.0.1:4321".parse().unwrap();
+        let server_id = ServerId::new(server_socket);
+
+        let mut secret_key = base.clone();
+        secret_key.extend_from_slice(&["--secret-key", "very secret"]);
+        assert_args_parse_to_auth(
+            secret_key,
+            AuthSubcommand::JwtHS256ServerToken {
+                server_id,
+                secret_key: "very secret".to_owned(),
+            },
+        );
+
+        // TODO: need some sample config to be used here
+        // let mut config = base;
+        // config.extend_from_slice(&["--config", "some/path.toml"]);
+    }
+
+    #[test]
+    fn auth_generate_shared_token_good() {
+        let raw_to_expected_bit_vals = &[
+            ("64", 64 / 8),
+            ("128", 128 / 8),
+            ("136", 136 / 8),
+            ("4000", 4_000 / 8),
+        ];
+
+        for (raw, expected) in raw_to_expected_bit_vals {
+            let args = auth_generate_shared_tokens_bits(raw);
+            assert_args_parse_to_auth(
+                args,
+                AuthSubcommand::Base64 {
+                    num_bytes: *expected,
+                },
+            );
+        }
+    }
+
+    // TODO: add tests for with sample config file
+    // #[test]
+    // fn scheduler_good() {
+    //     todo!()
+    // }
+
+    // TODO: add tests for with sample config file
+    // #[test]
+    // fn server_good() {
+    //     todo!()
+    // }
```

Comment:
Let's move this to a Github issue, and remove the comments from this file.

Reply:
That sound good to me. There's already quite a lot going on with this PR